### PR TITLE
Spring Cleaning

### PR DESCRIPTION
**Pull request by Benny-Nottonson**
Just some minor changes to some utils to clean up old code and get a little bit more performance :)

### CODE REVIEW

**Review by Benny-Nottonson — COMMENTED** [review]
I think using `[Type: DType]` in a function signature is definitely best practice if we have a global variable named dtype and we aren't asserting that `dtype == Type` although generally it is implied. With values I agree that we can use snake_case even for parameters, Types is a different story imho.

**Comment by StijnWoestenborghs** [comment]
Also note that it should be confused with a type parameter. Connected to a trait. 
```fn type parameter	fn do_it[Action: Actionable](action: Action)	PascalCase```
Which I agree should be capital

**Comment by StijnWoestenborghs** [comment]
I'm very ok with the changes though. Nice cleanup on the perf utils. That made it much more readable.
Did you say you implemented that proposed init method in Tensor here as well by @fnands? `__init__(self, owned existing: MojoTensor)` I also think that is a good idea (maybe you forgot to push it)

This also made me realize again we have that backward SUB operation in MNIST that takes 87% of the total backward time. Probably caused by de-broadcasting. We should have a look at that at some point.
```14   | SUB            | 0.2663              | 87.453 % ```

**Comment by Benny-Nottonson** [comment]
I did forget to commit, good catch

File: basalt/nn/tensor.mojo [hunk 1]
```diff
@@ -126,6 +133,11 @@ struct Tensor[dtype: DType](Stringable, Movable, CollectionElement):
         self._data = data
         self._shape = shape
 
+    @always_inline("nodebug")
+    fn __init__(inout self, owned tensor: _Tensor[dtype]):
```
Comment (fnands) [inline basalt/nn/tensor.mojo:137]: 🥳

**Review by StijnWoestenborghs — APPROVED** [review]

**Pull request merged**